app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  # Custom method: Updates the posts counter for a user
+
+  def update_user_posts_counter
+    user.update(post_counter: user.posts.count)
+  end
+
+  # Custom method: Returns the 5 most recent comments for a given post
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  # Custom method: Updates the likes counter for a post
+  def update_post_likes_counter
+    update(likes_counter: likes.count)
+  end
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20returns%20the%203%20most%20recent%20posts%20for%20a%20given%20user.), The `user` model should have one method `recent posts` which is A method that returns the 3 most recent posts for a given user.

[Hint] It will be something like this:
```ruby
def recent_posts
  posts.order('created_at Desc').limit(3)
end
```